docs/how-to/troubleshooting.md

```diff
@@ -1,173 +1,267 @@
+
+## Troubleshooting providing
+
+In this section, you will learn to troubleshoot common issues with providing. For a more detailed overview of the providing process, see [the lifecycle of data in IPFS](../concepts/lifecycle.md#2-providing).
+
+If clients are unable to find providers for a CID, the issue may lie in the content providing lifecycle, specifically _reprovider runs_, the continuous process in which a node advertises provider records. _Provider records_ are mappings of CIDs to network addresses, and have an expiration time of 48 hours, which accounts for provider churn.
+
+Generally speaking, as more files are added to an IPFS node, the longer reprovide runs take. When a reprovide run takes longer than 48 hours (the expiration time for provider records), CIDs will no longer be discoverable.
 
-At the top, you can see that this goroutine (number 2306090) has been waiting to acquire a semaphore for 458 minutes. That seems bad. Looking at the rest of the trace, we see the exact line it's waiting on is line 47 of runtime/sema.go. That's not particularly helpful, so we move on. Next, we see that call was made by line 205 of yamux/session.go in the `Close` method of `yamux.Session`. This one appears to be the issue.
+### Troubleshooting providing with Kubo
 
-Given that information, look for another goroutine that might be holding the semaphore in question in the rest of the stack dump.
+With this in mind, if no providers are returned, do the following:
 
-There are a few different reasons that goroutines can be hung:
+1. First, determine how long a reprovide run takes:
 
-- `semacquire` means we're waiting to take a lock or semaphore.
-- `select` means that the goroutine is hanging in a select statement, and none of the cases are yielding anything.
-- `chan receive` and `chan send` are waiting for a channel to be received from or sent on, respectively.
-- `IO wait` generally means that we are waiting on a socket to read or write data, although it *can* mean we are waiting on a very slow filesystem.
+   ```shell
+   ipfs stats reprovide
+   ```
 
-If you see any of those tags _without_ a `, X minutes` suffix, that generally means there isn't a problem -- you just caught that goroutine in the middle of a short wait for something. If the wait time is over a few minutes, that either means that goroutine doesn't do much, or something is pretty wrong.
+   The output should look something like this:
 
-If you see a lot of goroutines, consider using [stackparse](https://github.com/whyrusleeping/stackparse) to filter, sort, and summarize them.
+   ```shell
+   TotalReprovides:       4k (4,140)
+   AvgReprovideDuration:  12.870309s
+   LastReprovideDuration: 44m32.507618s
+   LastReprovide:         2025-06-10 09:45:18
+   ```
 
-### Analyzing the CPU Profile
+2. Note the value for `LastReprovideDuration`. If it is close to 48 hours, select one of the following options, keeping in mind that each has tradeoffs:
```

Comment:
:thought_balloon: depending on the strategy, the field right now will show how long a batch took, not the whole thing. I think its fine to keep the text here as-is – it will match reality once we have landed reprovide refactors in 0.36 and 0.37, however for now lets add this:

```suggestion
2. Note the value for `LastReprovideDuration`. If it is close to 48 hours, or if you notice a "reprovide taking too long" warning in your `ipfs daemon` output log, select one of the following options, keeping in mind that each has tradeoffs:
```